jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBCHandleConfiguration.java

```diff
@@ -214,6 +208,107 @@ public void run() {
         log.trace("JDBC source created");
         return true;
     }
+
+    /**
+     * Method used to create the query and publish handlers
+     * @param generalConfig     The general configuration of the JDBC Source
+     * @return                  Returns the maximum pool size, equal to twice the number of active tasks.
+     *                          If default active tasks is used, then returns 0.
+     */
+    private int createQueryAndPublishHandlers(Map<String, ?> generalConfig) {
+        int maxPoolSize = 0;
+
+        // Checking if asynchronous processing was specified in the general configuration
+        if (generalConfig.get(ASYNCH_PROCESSING) instanceof Boolean && (Boolean) generalConfig.get(ASYNCH_PROCESSING)) {
+            asynchronousProcessing = true;
+            int maxActiveTasks = MAX_ACTIVE_TASKS;
+            int maxQueuedTasks = MAX_QUEUED_TASKS;
+
+            if (generalConfig.get(MAX_ACTIVE) instanceof Integer && (Integer) generalConfig.get(MAX_ACTIVE) > 0) {
+                maxActiveTasks = (Integer) generalConfig.get(MAX_ACTIVE);
+
+                // Used to set the max pool size for connection pool
+                maxPoolSize = 2*maxActiveTasks;
```

Comment:
This sets `maxPoolSize` only if the user provides a `maxActiveTasks` setting.  Seems like we should have a default here?  Probably the same `2 * maxActiveTasks` is fine, but should be set regardless.  Zero's applicable only in the non-async case, I think.  Zero is probably never a valid pool size value.

(This is why we need the tests for taking the defaults in the configuration -- if not provided, it'll either fail or use some default that's not what we expect.)

Reply:
The reason I only set the value here is because my intention was to match whatever `2 * maxActiveTasks` would be. In this case, we make sure that we will have enough connections in our connection pool to match exactly twice the number of `maxActiveTasks` that was specified in the Source Configuration. Otherwise, our default value for `maxActiveTasks` is 5, and the default max of connections in the connection pool is 10 (so we don't have to set it in `JDBC.setupJDBC()`).

Let me know if that makes sense. I should add a comment inline to explain this behavior.

In any case, I am adding tests for the defaults now.